Integrations/integration-symantec-DLP-incident-reportintg-update.yml

```diff
@@ -0,0 +1,336 @@
+
+
+    def incidentBinaries(args):
+
+        return client.service.incidentBinaries(**args)
+
+
+    def myconverter(o):
+        if isinstance(o, datetime):
+            return o.__str__()
+
+
+    def fetchincidents():
+        last_run=demisto.getLastRun()
+        if 'lt_time' not in last_run:
+            fetch_time,_ = parse_date_range(FETCH_TIME)
+            last_run['lt_time'] = fetch_time
+        else:
+            last_run['lt_time']= datetime.strptime(last_run['lt_time'],'%d/%m/%Y %H:%M:%S.%f')
+
+        incidentids=get_incident_id(last_run['lt_time'])
+        incidents=[]
+        if len(incidentids['incidentId']) > 0:
+            for incident1 in incidentids['incidentId']:
+                inc=incidentDetail(str(incident1))
+                serialized_obj = helpers.serialize_object(inc[0])
+                incidents.append({
+                    'name':"#"+str(incident1) + " DLP "+ inc[0]['incident']['violatedPolicyRule'][0]['ruleName']+ "RULE has been violated",
+                    'details': json.dumps(serialized_obj,default = myconverter),
+                    'rawJSON': json.dumps(serialized_obj,default = myconverter),
+                    'severity' : SEVERITY[inc[0]['incident']['severity']]
+                })
+                if inc[0]['incident']['incidentCreationDate'].isoformat() > last_run['lt_time'].isoformat():
+                    demisto.setLastRun({'lt_time':inc[0]['incident']['incidentCreationDate'].strftime("%d/%m/%Y %H:%M:%S.%f")})
+        demisto.incidents(incidents)
+
+
+    def updateIncidents():
+        args = demisto.args()
+        inc_id = args['incidentId']
+        args.pop('incidentId')
+        if args['status'] == 'New':
+            args['status'] = 'incident.status.New'
+        return client.service.updateIncidents(updateBatch={'batchId':str(random.randint(11111,99999)),'incidentId':inc_id,'incidentAttributes':args})
+
+
+
+    if demisto.command() == 'fetch-incidents':
+        fetchincidents()
+
+    elif demisto.command() =='test-module':
+        try:
+            inc_list = get_incident_id(datetime.now())
+            demisto.results("ok")
+        except:
+            return_error("Invalid instance details")
+
+    elif demisto.command() == 'sdlp-incident-details':
+        inc_details = incidentDetail( demisto.args()['incidentId'])
+        res = helpers.serialize_object(inc_details[0])
+        json_res = json.loads(json.dumps(res,default = myconverter))
+        demisto.results({'ContentsFormat': formats['json'],
+            'Type': entryTypes['note'],
+            'Contents':json_res,
+            'EntryContext':{'SymantecDLP':{'Incident':json_res}}}
+        )
+
+    elif demisto.command() == 'sdlp-update-incident':
+        demisto.results(updateIncidents())
+
+    elif demisto.command() == 'sdlp-incident-binaries':
+        args = demisto.args()
+        binaries = incidentBinaries(args)
+        if len(binaries["Component"]) > 0 :
```

Comment:
This is a bit risky, as if `Component` is not in `binaries`, a KeyError Exception will be thrown
Please use `.get()`